spec/abci++/abci++_basic_concepts_002_draft.md

```diff
@@ -42,50 +61,75 @@ called, methods `PrepareProposal`, `ProcessProposal`, `ExtendVote`, and
+This has a negative impact on Tendermint's liveness, i.e., if vote extensions repeatedly cannot be
+verified by correct validators, Tendermint may not be able to finalize a block even if sufficiently
+many (+2/3) validators send precommit votes for that block. Thus, `VerifyVoteExtension`
+should be used with special care.
 As a general rule, an Application that detects an invalid vote extension SHOULD
-accept it in `ResponseVerifyVoteExtension` and ignore it in its own logic. Tendermint calls it when 
+accept it in `ResponseVerifyVoteExtension` and ignore it in its own logic. Tendermint calls it when
 a process receives a precommit message with a (possibly empty) vote extension.
 
-* [**FinalizeBlock:**](./abci++_methods_002_draft.md#finalizeblock) It delivers a decided block to the Application. The Application must execute the transactions in the block in order and update its state accordingly. Cryptographic commitments to the block and transaction results, via the corresponding
-parameters in `ResponseFinalizeBlock`, are included in the header of the next block. Tendermint calls it when a new block is decided.
+- [**FinalizeBlock:**](./abci++_methods_002_draft.md#finalizeblock) It delivers a decided block to
+the Application. The Application must execute the transactions in the block in order and update its
+state accordingly. Cryptographic commitments to the block and transaction results, returned via the
+corresponding parameters in `ResponseFinalizeBlock`, are included in the header of the next block.
+Tendermint calls it when a new block is decided.
+
+- [**Commit:**](./abci++_methods_002_draft.md#commit) Instructs the Application to persist its
+state. It is a fundamental part of Tendermint's crash-recovery mechanism that ensures the
+synchronization between Tendermint and the Applicatin upon recovery. Tendermint calls it just after
+having persisted the data returned by `ResponseFinalizeBlock`.
 
 ### Mempool methods
 
-* [**CheckTx:**](./abci++_methods_002_draft.md#checktx) This method allows the Application to validate transactions against its current state, e.g., checking signatures and account balances. If a transaction passes the validation, then tendermint adds it to its local mempool, discarding it otherwise. Tendermint calls it when it receives a new transaction either coming from an external user or another node. Furthermore, Tendermint can be configured to re-call `CheckTx` on any decided transaction (after `FinalizeBlock`).
+- [**CheckTx:**](./abci++_methods_002_draft.md#checktx) This method allows the Application to validate transactions against its current state, e.g., checking signatures and account balances. If a transaction passes the validation, then tendermint adds it to its local mempool, discarding it otherwise. Tendermint calls it when it receives a new transaction either coming from an external user or another node. Furthermore, Tendermint can be configured to call re-`CheckTx` on all
```

Comment:
```suggestion
- [**CheckTx:**](./abci++_methods_002_draft.md#checktx) This method allows the Application to validate transactions against its current state, e.g., checking signatures and account balances. If a transaction passes the validation, then Tendermint adds it to its local mempool, discarding it otherwise. Tendermint calls it when it receives a new transaction either coming from an external user or another node. Furthermore, Tendermint can be configured to call re-`CheckTx` on all
```

Reply:
Thanks for spotting